assets/tests/hooks/usePatternsByIdForRoute.test.ts

```diff
@@ -32,10 +34,12 @@ describe("usePatternsByIdForRoute", () => {
       return usePatternsByIdForRoute("66")
     })
     expect(mockFetchRoutePatterns).toHaveBeenCalledTimes(1)
-    expect(result.current).toEqual({ [rp1.id]: rp1, [rp2.id]: rp2 })
+    await waitFor(() =>
+      expect(result.current).toEqual({ [rp1.id]: rp1, [rp2.id]: rp2 })
+    )
   })
 
-  test("doesn't fetch routes when passed null id", () => {
+  test("doesn't fetch routes when passed null id", async () => {
     const mockFetchRoutePatterns: jest.Mock =
       Api.fetchRoutePatterns as jest.Mock
     mockFetchRoutePatterns.mockReturnValueOnce(
```

Comment:
Non-blocking: There are two more instances of `instantPromise` remaining in this file that could be replaced with `mockResolvedValue`.

Happy to do that in a follow-up instead of as part of this PR, but I did want to flag it.